### ISSUE TITLE
DotTip initial focus doesn't work any longer

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#8205]
**Describe the bug**

In Gutenberg 3.2.0 the **first** NUX "tip" used to get focused on mount. Setting initial focus within the first tip "popover" is essential for keyboard and screen reader users. Basically all the tips must behave like sort of a modal dialog and focusing the first one is important to make users able to use the tip.

**To Reproduce**
- revert to Gutenberg 3.2.0
- clear the local storage in your browsers in order to get again the "tips"
- create a new post
- on page load, when the first tip renders, it should get the initial focus on its `container` (see screenshot below)
- press the Tab key and you will see focus is on the tip and tabbing is constrained within the tip

The initial focus on the first tip in 3.2.0:

![screen shot 2018-07-25 at 18 59 28](https://user-images.githubusercontent.com/1682452/43219905-e9d225ce-9048-11e8-911a-1484ad3f498b.png)

- switch to Gutenberg current master
- repeat the steps above
- initial focus is not set on the first tip
- press Tab and you will see focus is on the first element in the page: tabbing starts normally from the skip link (see screenshot below)

![screen shot 2018-07-25 at 20 17 19](https://user-images.githubusercontent.com/1682452/43219983-1722775e-9049-11e8-9489-f392af4b2576.png)

Both the DotTip and the popover component have gone through some recent refactoring. Especially the popover `focusOnMount` prop has changed. Seems to me the real cause of this regression is about timings though.

Note: I've tested the other usages of popovers with `focusOnMount` and they all seem to work to me, so this issue seems specific to the DotTip.

To my understanding:
- DotTip was a class component and [used a setTimeout](https://github.com/WordPress/gutenberg/commit/c85e3f98bcfdf9dd8a687d0fa6b501f0dab79dc0#diff-664e6197e2d4352dfedbaa4411d4a0e2L21) to recalculate its position, refresh, and focus
- it doesn't use setTimeout any longer but `withSafeTimeout` is still imported and used, I guess it should be removed
- seems to me there's now the need to ensure the popover itself uses a safe timeout before calling `refresh()` and `focus()` on `componentDidMount()`

/Cc @aduth
